dotnet/Gherkin/GherkinDialectProvider.cs

```diff
@@ -55,8 +52,7 @@ protected virtual Dictionary<string, GherkinLanguageSetting> LoadLanguageSetting
 
     protected virtual Dictionary<string, GherkinLanguageSetting> ParseJsonContent(string languagesFileContent)
     {
-        // ReSharper disable once InvokeAsExtensionMethod
-        return JSONParser.FromJson<Dictionary<string, GherkinLanguageSetting>>(languagesFileContent);
+        return JsonSerializer.Deserialize<Dictionary<string, GherkinLanguageSetting>>(languagesFileContent, new JsonSerializerOptions(JsonSerializerDefaults.Web));

```

Comment:
For the Java implementation I removed the dependency on serialization by generating the source code for the Gherkin dialects file. You may want to consider a similar approach. Not necessarily in this pull request.

- https://github.com/cucumber/gherkin/blob/main/java/src/main/groovy/io/cucumber/gherkin/GherkinDialects.groovy
- https://github.com/cucumber/gherkin/blob/main/java/src/main/groovy/io/cucumber/gherkin/GherkinDialects.gsp

Reply:
That is reasonable, but then https://github.com/cucumber/messages/pull/237#issuecomment-2225649432 will have to be addressed.

Reply:
ok, I've added test for that

Reply:
Effectively yes.

The upgrade will fail, so Renovate will raise a PR instead. Then we close the PR and Renovate will take note and not offer the same upgrade again.

Reply:
And you can see it in action here https://github.com/cucumber/gherkin/pull/279

Reply:
Hm, it will update all nuget packages very time or 1 PR = 1 nuget package?

It could be a problem
https://github.com/cucumber/gherkin/pull/278

Reply:
It should be one PR for each dependency unless there is a a profile that bundles them. These profiles are manually written. So usually quite sane.